aiohttp/connector.py

```diff
@@ -1099,6 +1103,8 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
 
             host_resolved.add_done_callback(drop_exception)
             raise
+        except asyncio.TimeoutError:
+            raise
         except OSError as exc:
             # in case of proxy it is not ClientProxyConnectionError
             # it is problem of resolving proxy ip itself
```

Comment:
```suggestion
        except OSError as exc:
            if exc.errno is None and isinstance(exc, asyncio.TimeoutError):
                raise
            # in case of proxy it is not ClientProxyConnectionError
            # it is problem of resolving proxy ip itself
```